shell/config/router/navigation-guards/products.js

```diff
@@ -0,0 +1,15 @@
+import { setProduct } from '@shell/utils/auth';
+import { applyProducts } from '@shell/store/type-map';
+
+export function install(router, context) {
+  router.beforeEach((to, from, next) => loadProducts(to, from, next, context));
+}
+
+export async function loadProducts(to, from, next, { store }) {
+  // GC should be notified of route change before any find/get request is made that might be used for that page
+  store.dispatch('gcRouteChanged', to);
+
+  await applyProducts(store, store.$plugin);
+  setProduct(store, to);
```

Comment:
Sorry for the naive question, would not be better an action in the store for this?

Reply:
I'm happy to move it if you'd like. Is there a store that you think fits this best?

It seems a bit awkward for our existing store since it touches the gc stuff, products and plugins.

Reply:
Let's keep it minimal and think about it in another phase then, e.g. review of the store.